kuma/dashboards/views.py

```diff
@@ -48,6 +48,24 @@ def revisions(request):
                         datetime.datetime.now())
             query_kwargs['created__range'] = [start_date, end_date]
 
+        preceding_period = filter_form.cleaned_data['preceding_period']
+        if preceding_period:
+            # these are messy but work with timedelta's seconds format,
+            # and keep the form and url arguments human readable
+            if preceding_period == 'month':
+                seconds = 30 * 24 * 60 * 60
+            if preceding_period == 'week':
+                seconds = 7 * 24 * 60 * 60
+            if preceding_period == 'day':
+                seconds = 24 * 60 * 60
+            if preceding_period == 'hour':
+                seconds = 60 * 60
+            # use the form date if present, otherwise, offset from now
+            end_date = (filter_form.cleaned_data['end_date'] or
+                        datetime.datetime.now())
+            start_date = end_date - datetime.timedelta(seconds=seconds)
```

Comment:
Please use `django.utils.timezone.now()` here instead in preparation for enabling Django's timezone support.